src/main/java/de/tum/in/www1/artemis/config/websocket/WebsocketConfiguration.java

```diff
@@ -262,12 +265,26 @@ public Message<?> preSend(@NotNull Message<?> message, @NotNull MessageChannel c
 
         /**
          * Returns whether the subscription of the given principal to the given destination is permitted
+         * Database calls should be avoided as much as possible in this method.
+         * Only for very specific topics, database calls are allowed.
          *
          * @param principal   User principal of the user who wants to subscribe
          * @param destination Destination topic to which the user wants to subscribe
          * @return flag whether subscription is allowed
          */
         private boolean allowSubscription(Principal principal, String destination) {
+            if (isBuildQueueAdminDestination(destination)) {
```

Comment:
please additionally add a comment in the beginning of the method, developers typically do not read the JavaDoc above :-(